src/rgw/setup/rgw.py

```diff
@@ -1059,6 +1059,19 @@ def _update_resource_limit_based_config(conf: MappedConf, client_section: str):
         confstore_url = const.CONFSTORE_FILE_HANDLER + svc_config_file
         Rgw._load_rgw_config(Rgw._conf_idx, confstore_url)
 
+        concurrent_max_requests_val = conf.get(const.GCONF_MAX_CONCURRENT_REQ_KEY)
+        thread_pool_size_val = conf.get(const.GCONF_THREAD_POOL_SIZE_KEY)
+
+        # check if user specified these values explicitly.
+        # if not then populate formula based values.
+        if (concurrent_max_requests_val is not None and thread_pool_size_val is not None):
+           Conf.set(Rgw._conf_idx, f'{client_section}>{const.SVC_CONCURRENT_MAX_REQ_KEY}',
```

Comment:
what if only one of the value is user supplied ? handle separately
Also what about validation of these values against minimum values ?

Reply:
if one of the value is provided in solution.yaml file then below code will use resource based formula and generate the values . line 1106 & 1109 

Regarding validation of these values, as of now we are not doing any validation for any config keys specified in solution.yaml. 
we have validation only on max resource limit keys i.e. 
`
limits:
      num_services: 1
      services:
      - name: rgw
        memory:
          min: 128Mi
          max: 2Gi
        cpu:
          min: 250m
          max: 1000m
`

apart form this we dont do any validation on values provided in soultion.yaml as extra_configuration.

Reply:
> if one of the value is provided in solution.yaml file then below code will use resource based formula and generate the values . line 1106 & 1109

Yes. Am saying either here we can handle here or down below lines, to set config from solution.yaml or using formula and better if we handle individually per config.

and if both set from user i/p, then yes return from here

Reply:
after our discussion, i see the need to keep these checks separate still, agree to changes